docs/hu/SQL_Cheat_Sheet_magyar.md

```diff
@@ -0,0 +1,471 @@
+
+collection.Find(x => x.Price != 123);
+collection.Find(Builders<Product>.Filter.Ne(x => x.Price, 123)); // Ne, mint not equals
+
+collection.Find(x => x.Price >= 123);
+collection.Find(Builders<Product>.Filter.Gte(x => x.Price, 123)); // Gte, mint greater than or equal to
+
+collection.Find(x => x.Price < 123);
+collection.Find(Builders<Product>.Filter.Lt(x => x.Price, 123)); // Lt, mint less than
+
+collection.Find(x => x.Price < 500 || x.Stock < 10);
+collection.Find(
+    Builders<Product>.Filter.Or(
+        Builders<Product>.Filter.Lt(x => x.Price, 500),
+        Builders<Product>.Filter.Lt(x => x.Stock, 10)
+    )
+);
+
+collection.Find(x => !(x.Price < 500 || x.Stock < 10));
+collection.Find(
+    Builders<Product>.Filter.Not(
+        Builders<Product>.Filter.Or(
+            Builders<Product>.Filter.Lt(x => x.Price, 500),
+            Builders<Product>.Filter.Lt(x => x.Stock, 10)
+        )
+    )
+);
+```
+
+Példa .Group() második paraméterében lehet vetítéseket megadni hasonlóan a .Project() függvényhez.
+```csharp
+var r=productsCollection.Aggregate()
+    .Group(
+        //csoportosítás rész
+        p=>p.VAT.Percentage,
+        //projekcios rész
+        p=>new{
+        vatp=p.Key
+        sumPrice=p.Sum(s=>s.Price)
+    })
+    .ToList();
+
+```
+
+## Order By
+**SQL:**
+
+```SQL
+ORDER BY column1 ASC, column2 DESC;
+```
+
+**C# LINQ:**
+
+Kétszintű rendezés
+```csharp
+.OrderBy(item => item.Column1)
+.ThenByDescending(item => item.Column2)
+```
+**C# MongoDb:**
+
+```csharp
+.Sort(Builders<CollectionItem>.Sort.Ascending(item=>item.Coloumn))
+```
+
+## JOIN
+
+**SQL:**
+
+```SQL
+SELECT table1.column, table2.column
+FROM table1
+JOIN table2 ON table1.column = table2.column;
+```
+
+**C# LINQ:**
+
+Mivel Navigation propertyk segítségével általában elérhetőek az asszociált osztályok, explicit illeszteni csak akkor kell, ha a feladat megoldási logikája megkívánja vagy nincs navigation property az egyik osztályban a másikra.
+```csharp
+var query = context.Table1
+            .Join(context.Table2,
+                  item1 => item1.Column,
+                  item2 => item2.Column,
+                  (item1, item2) => new { item1.Column, item2.Column });
+//SQL szerű LINQ szintaktikával:
+var query = from item1 in context.Table1
+            join item2 in context.Table2
+            on item1.Column equals item2.Column
+            select new { item1.Column, item2.Column };
+```
+**Megjegyzés:** Innentől C# LINQ esetén csak a kényelmesebb, query szintatkiai jelölés lesz látható.
+
+**C# MongoDb:**
+
+**MongoDb-ben nem tanultuk szerveri oldari joinra módszert. LINQ segítségével van az után, hogy a teljes illesztendő adathalmazokat kiolvastuk az adatbázisból.** Az adatok lekérdezése után általában a .ToHashSet() és a .Contains() metódusok segítségével, kliens oldali dictionary készítéssel végezzük el az illesztést. (Lásd MongoDb gyakorlat 1. Feladat 5. pontja)
```

Comment:
egyébként nincs is az új mongodb-ben erre lehetőség? utána tudnál nézni ennek kérlek?

Reply:
Albert István Tanár Úr hívta fel rá a figyelmemet, hogy "mongodbnél is van szerver join szerűség, csak nem tanultuk, legyünk pontosak." még korábban emailben. Most utánanéztem talán olyasmit lehet csinálni, hogy egy viewt hozunk létre ami összefogja a két collectiont egybe. Másik opciónak találtam a $lookup staget, bár pontosan talán nem sikerült megértenem még, hogy hogyan működik. (https://www.mongodb.com/docs/manual/reference/operator/aggregation/lookup/)